### DISCUSSION TITLE
How do I prevent Controller from registering hidden inputs?

### DISCUSSION THREAD

**Discussion by Lori-Becker** [react-hook-form/react-hook-form#4202] [Q&A] [answered]
I'm using Controller with some Material-UI inputs. Some of these inputs only get displayed if a user enters certain values in other inputs. When I run DevTools, it shows these hidden inputs as registered. When the form is submitted, these hidden inputs fail validation. I don't want inputs to be registered unless they are visible. How do I prevent Controller from registering hidden inputs?

**Comment by bluebill1049** [comment, marked as answer]
I think you can use native `register` for native hidden inputs right? i think a simple solution would be unmount that input, if you don't want to be included in the validation.
